### PR TITLE
feat: working basic bytecode

### PR DESCRIPTION
**Pull request by sno2**
Depends on #69  and #64

### CODE REVIEW

**Comment by aapoalas** [comment]
Awesome! <3

File: nova_vm/src/engine/bytecode/executable.rs [hunk 1]
```diff
@@ -123,14 +125,80 @@ impl Compile for ast::NumberLiteral<'_> {
...
+        });
+        ctx.exe
+            .add_instruction_with_constant(Instruction::StoreConstant, constant);
+    }
+}
+
+impl Compile for ast::UnaryExpression<'_> {
+    fn compile(&self, ctx: &mut CompileContext) {
+        match self.operator {
+            // 13.5.5.1 Runtime Semantics: Evaluation
```
Comment (aapoalas) [inline nova_vm/src/engine/bytecode/executable.rs:152]: <3

File: nova_vm/src/ecmascript/execution/environments.rs [hunk 2]
```diff
@@ -44,34 +125,136 @@ impl Default for Environments {
...
+            // [[Strict]]: strict,
+            strict,
+            // [[ThisValue]]: EMPTY
+            this_value: None,
+        });
+        // }.
+    };
+
+    // 2. Let exists be ? env.HasBinding(name).
+    let exists = match env {
```
Comment (sno2) [inline nova_vm/src/ecmascript/execution/environments.rs:157]: Should we reproduce every Environment table method on `EnvironmentIndex` which would find the respective method to call?

File: nova_vm/src/ecmascript/execution/environments/declarative_environment.rs [hunk 3]
```diff
@@ -1,56 +1,285 @@
...
-// often not needed by algorithms.
 #[derive(Debug)]
-pub struct Binding {
-    pub value: Option<Value>,
-    pub strict: bool,
-    pub mutable: bool,
-    pub deletable: bool,
+struct Binding {
+    value: Option<Value>,
+    // TODO: Pack these into bitfields.
```
Comment (sno2) [inline nova_vm/src/ecmascript/execution/environments/declarative_environment.rs:29]: Aapo - will include the new TODO change once #64 is merged.

**Review by aapoalas — APPROVED** [review]

**Pull request merged**